File: 프로그래머스/1/68935. 3진법 뒤집기/3진법 뒤집기.cpp
#include <string>
#include <vector>
#include <iostream>
#include <string>
#include <algorithm>
#include <cmath>
using namespace std;

int solution(int n) {
    int answer = 0;
    string data = "";
    
    while(n>0)
    {
        int rest = n%3;
        n /= 3;
        data+= rest;
    }
    
    
    int stringSize = data.size();
    reverse(data.begin(),data.end());
    for(int i=0;i<stringSize;++i)
    {
        int dData = data[i];
        answer += dData * pow(3,i);
    }
    
    
    return answer;
}

File: 프로그래머스/lv1/68935. 3진법 뒤집기/3진법 뒤집기.cs
using System;
using System.Text;
public class Solution {
    public int solution(int n) {
        StringBuilder sb = new StringBuilder(200);
        int answer = 0;
        int tmpN = n;
        int rest = 0;
        int restoreTens = 0;
        
        while(tmpN>=1)
        {
            rest = tmpN%3;
            tmpN /= 3;
            sb.Append(rest);
        }
        char[] changed = sb.ToString().ToCharArray();
        Array.Reverse(changed);
        
        for(int i=0;i<changed.Length;++i)
        {
            int value = (int)Math.Pow(3,i)*(changed[i]-'0');
            answer += value;
        }
        
        
        return answer;
    }
}